designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,213 @@
+      registry: docker.io
+      repository: ""
+      tag: latest
+      pullPolicy: IfNotPresent
+
+     ## The environment variables of the controller container.
+     ## More info: https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/
+  	env: []
+
+    ## Configure extra options for liveness probe.
+    ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
+    livenessProbe:
+      enabled: true
+      httpGet:
+        path: /healthz
+        port: 8081
+      initialDelaySeconds: 15
+      periodSeconds: 20
+
+    readinessProbe:
+      enabled: true
+      httpGet:
+        path: /readyz
+        port: 8081
+      initialDelaySeconds: 5
+      periodSeconds: 10
+
+    ## Configure the resources.
+    ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+    resources:
+      limits:
+        cpu: 500m
+        memory: 128Mi
+      requests:
+        cpu: 10m
+        memory: 64Mi
+
+     ## Configure the scheduling rules.
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+     affinity: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
+     nodeSelector: {}
+
+     ## More info: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration
+     tolerations: {}
+
+     ## Configure the rbac roles.
+     serviceAccountName: ""
+
+## Configure the webhook
+webhook:
+  enabled: true
+  ## More info: https://kubernetes.io/docs/concepts/services-networking/service/
+  service:
+    type: ClusterIP
+    ports:
+    - port: 443
+      protocol: TCP
+      targetPort: 9443
+
+metrics:
+  enabled: true
+  service:
+    type: ClusterIP
+    ports:
+    - port: 8443
+      protocol: TCP
+
+## Configure the cert-manager
+certmanager:
+  enabled: true
+  installCRDs: true
+  ## Add the owner reference to the certificate.
+  extraArgs:
+  - --enable-certificate-owner-ref=true
+
+prometheus:
+  ## Enable the Prometheus Monitor
+  enabled: false
+
+## The domain name of Kubernetes cluster 
+## More info: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/
+clusterDomain: cluster.local
+```
+
+### User Stories
+
+- As a developer, I want to be able to generate a helm chart from a kustomize directory so that I can distribute the helm chart to my users. Also, I want the 
+generation to be as simple as possible without the need to write any additional duplicate files.
+
+- As a user, I want the helm chart can cover all potential configurations when I deploy it on the Kubernetes cluster.
+
+### Implementation Details/Notes/Constraints [optional]
+
+TBF
+
+### Risks and Mitigations
+
+TBF
+
+### Proof of Concept [optional]
+
+Refer to the open source tool
+[helmify](https://github.com/arttor/helmify).
+
+
+## Drawbacks
+
+TBF
+
+## Alternatives
+
+TBF
```

Comment:
Here I think we need disccuss the command option 
ALso, add the pros and cons of each one of them

Reply:
Can you just add the command here ? Then that is fine